src/block.rs

```diff
@@ -15,13 +18,16 @@ pub enum BlockError {
 pub type BlockResult<T> = core::result::Result<T, BlockError>;
 
 /// Represent a certain amount of data from a block device.
-#[derive(Clone)]
+#[derive(Clone, Copy)]
 #[repr(C, align(2))]
 pub struct Block {
```

Comment:
Call this block `AhciBlock` and document it as such. 

Maybe create a bunch of other useful boilerplate blocks with no alignment, like `Block512`, `Block1024`, `Block4096`.

Reply:
🤔  I'm wondering whether this should be moved out of storage-device entirely.

Reply:
I'm definitely deleting this. The more I think about it, the less I can find a reasonable reason to leave this implementation in storage-device.

Reply:
Block512/Block1024/Block4096 would be sort of pointless, as ideally you'd just pass in an [u8; 4096]. Except, of course, you can't do that, because arrays don't implement all the necessary traits. *sigh*.

I'm of the opinion that we shouldn't provide any block implementation. I'll add an example to the Block type how to write a block implementation.